src/agent/src/StructuredOutput/AgentProcessor.php

```diff
@@ -77,6 +80,11 @@ public function processInput(Input $input): void
             throw new InvalidArgumentException('Streamed responses are not supported for structured output.');
         }
 
+        $modelObject = $this->platform->getModelCatalog()->getModel($input->getModel());
+        if (!\in_array(Capability::OUTPUT_STRUCTURED, $modelObject->getCapabilities(), true)) {
+            throw MissingModelSupportException::forStructuredOutput($modelObject->getName());
```

Comment:
this would be enough for me
```suggestion
        $model = $this->platform->getModelCatalog()->getModel($input->getModel());
        if (!\in_array(Capability::OUTPUT_STRUCTURED, $model->getCapabilities(), true)) {
            throw MissingModelSupportException::forStructuredOutput($model->getName());
```